storage/scan_queue.go

```diff
@@ -118,15 +119,124 @@ func (sq *scanQueue) shouldQueue(now time.Time, rng *Range) (shouldQ bool, prior
 // intents are older than intentAgeThreshold. The very act of scanning
 // keys verifies on-disk checksums, as each block checksum is checked
 // on load.
-func (sq *scanQueue) process(now time.Time, rng *Range) error {
+func (sq *scanQueue) process(now proto.Timestamp, rng *Range) error {
 	snap := rng.rm.Engine().NewSnapshot()
 	iter := newRangeDataIterator(rng, snap)
 	defer iter.Close()
 	defer snap.Stop()
 
+	// Lookup the GC policy for the zone containing this key range.
+	policy, err := sq.lookupGCPolicy(rng)
+	if err != nil {
+		return err
+	}
+
+	scanMeta := proto.NewScanMetadata(now.WallTime)
+	gc := engine.NewGarbageCollector(now, policy)
+
+	// Compute intent expiration (intent age at which we attempt to resolve).
+	intentExp := now
+	intentExp.WallTime -= intentAgeThreshold.Nanoseconds()
+
+	gcArgs := &proto.InternalGCRequest{
+		RequestHeader: proto.RequestHeader{
+			Key:       rng.Desc.StartKey,
+			Timestamp: now,
+			RaftID:    rng.Desc.RaftID,
+		},
+	}
+	var expBaseKey proto.Key
+	var keys []proto.EncodedKey
+	var vals [][]byte
 	for ; iter.Valid(); iter.Next() {
-		// TODO(spencer): implement processing.
+		baseKey, ts, isValue := engine.MVCCDecodeKey(iter.Key())
+		if !isValue {
+			// Moving to the next key (& values). If there's more than a
+			// single value for the key, possibly send for GC.
+			if len(keys) > 1 {
```

Comment:
Well single valued keys would still show up as len(keys) > 1 in that case. Because these are the raw, encoded keys, so there's one for the metadata and one for the deletion tombstone with timestamp.